pxemgr/etcd_discovery_handlers.go

```diff
@@ -104,11 +106,28 @@ func (mgr *pxeManagerT) etcdDiscoveryProxyRequest(r *http.Request) (*http.Respon
 	if err != nil {
 		return nil, err
 	}
+	var transport = http.DefaultTransport
+
+	if strings.HasPrefix(mgr.etcdEndpoint, "https") && mgr.etcdCAFile != "" {
+		customCA := x509.NewCertPool()
+
+		pemData, err := ioutil.ReadFile(mgr.etcdCAFile)
+		if err != nil {
+			return nil, errors.New("Unable to read custom CA file: "+err.Error())
+		}
+		customCA.AppendCertsFromPEM(pemData)
+		transport = &http.Transport{
+			TLSClientConfig: &tls.Config{RootCAs:customCA},
+		}
+	}
+
+	scheme := strings.Split(mgr.etcdEndpoint,":")[0]
+	host := strings.Split(mgr.etcdEndpoint,"/")[2]
```

Comment:
You can use https://golang.org/pkg/net/url/#Parse and directly use the result instead of this manual magic.